### ISSUE TITLE
Use a .grain extension for grain backups

### ISSUE THREAD

**Issue by ocdtrekkie** [sandstorm-io/sandstorm#1274]
I kinda find it confusing when my Sandstorm grain backups aren't obviously sortable in my downloaded files, or can blend in with other things I'm working with. Most users will probably not do a lot of grain modification, and those that do will probably be able to discover or read that it can be renamed and opened as a zip file. (Like Android APK files.)

A .grain filetype would clearly denote my backups as grains, just like the .spk filetype clearly denotes app packages. This would also help when restoring grains, because the browser would be able to only display Sandstorm .grains, not all zip files the user might have.

**Comment by kentonv** [comment]
Yeah I've sort of wanted to do this too.

**Comment by zarvox** [comment]
One downside is that for backwards-compatibility with older backups, the upload backup filepicker still has to allow selecting `.zip` files.  Alternately we need to document the change REALLY well, and give people instructions for converting their old backups to something that will work with newer `.grain`-requiring Sandstorm instances. :/

**Comment by ocdtrekkie** [comment]
Combine documenting it really well (should be done anyways) ;) with allowing both for an extended period of time? If you wanna change it at all, the sooner the better while the number of affected users is smaller.

**Comment by kentonv** [comment]
Is it actually possible to instruct a web file picker to show only files with certain extensions? I thought it wasn't (making Jake's last sentence in the initial bug description impossible), but maybe that has changed recently.

**Comment by zarvox** [comment]
Absolutely: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#Attributes (see the "accept" attribute)

We should probably be doing that for the `.spk` upload input as well :)

**Comment by paulproteus** [comment]
cc: @jparyani
